doc/source/server_side_encryption.rst

```diff
@@ -0,0 +1,62 @@
+.. _server_side_encryption:
+
+########################
+Server-Side Encryption
+########################
+
+***********************
+Cloud
+***********************
+B2 cloud supports `Server-Side Encryption <https://www.backblaze.com/b2/docs/server_side_encryption.html>`_. All read
+and write operations provided by **b2sdk** accept encryption settings as an optional argument. Not supplying this
+argument means relying on bucket defaults - for **SSE-B2** and for no encryption. In case of **SSE-C**, providing an
+encryption key is crucial for successful downloading and copying.
+
+***
+API
+***
+Methods and classes that require encryption settings all accept an `EncryptionSetting` object, which holds information
+about present or desired encryption (mode, algorithm, key, key_id). Some, like copy operations, accept two sets of settings (for
+source and for destination). Sync, however, accepts an `EncryptionSettingsProvider` object, which is an
+`EncryptionSetting` factory, yielding them based on:
+
+* a :class:`b2sdk.v1.Bucket` object
+* file id
+* file name
+* file info - for files already existing in B2 cloud
```

Comment:
hold on, can't I select a key based on file_info such as where `{"Author": "Mary"}` I will use `my_keys.get[file_info["Author"]]`? I think this should be very allowed.